mesonbuild/compilers/compilers.py

```diff
@@ -1097,16 +1095,17 @@ def _get_compile_output(self, dirname, mode):
         return os.path.join(dirname, 'output.' + suffix)
 
     @contextlib.contextmanager
-    def compile(self, code, extra_args=None, mode='link', want_output=False):
+    def compile(self, code, extra_args=None, mode='link', want_output=False, cdata: coredata.CoreData = None):
```

Comment:
the type is invalid, you've assigned None but the type says it must be a `coredata.CoreData` instance. Either this needs to be a positional argument or `typing.Optional[coredata.CoreData]`